internal/backend/remote-state/s3/client.go

```diff
@@ -648,6 +653,20 @@ func (c *RemoteClient) configurePutObjectACL(i *s3.PutObjectInput) {
 	i.ACL = types.ObjectCannedACL(c.acl)
 }
 
+func (c *RemoteClient) configurePutObjectTags(i *s3.PutObjectInput) {
+	if len(c.tags) == 0 {
+		return
+	}
+	tagPairsSlice := make([]string, len(c.tags))
+	index := 0
+	for k, v := range c.tags {
+		tagPairsSlice[index] = fmt.Sprintf("%s=%s", url.QueryEscape(k), url.QueryEscape(v))
```

Comment:
You could try using `url.Values` since the [`Encode`](https://pkg.go.dev/net/url#Values.Encode) method on that do already handle the escaping and it builds the content already as a URL query string.

Reply:
Much easier, thanks for the pointer